### PR TITLE
Feat/import jobs

### PR DESCRIPTION
**Pull request by guillaumeprevost76**
1st step of import works.
It contains the import function for jobs based on the json format created by export function.

### CODE REVIEW

File: saagieapi/jobs/jobs.py [hunk 1]
```diff
@@ -892,3 +892,83 @@ def export(
             logging.warning("❌ Job [%s] has not been successfully exported", job_id)
             write_error(error_folder, "jobs", job_id)
         return result
+
+    def import_job(
```
Comment (tquiviger) [inline saagieapi/jobs/jobs.py:896]: Rename to `import` and not `import_job`

File: saagieapi/jobs/jobs.py [hunk 2]
```diff
@@ -892,3 +892,83 @@ def export(
...
+            Path to the package of the job to import
+        Returns
+        -------
+        bool
+            True if job is imported False otherwise
+        """
+        result = True
+
+        try:
+            jobId = job_info["id"]
```
Comment (tquiviger) [inline saagieapi/jobs/jobs.py:920]: all variables should be in snake_case and not camel_case

**Review by tquiviger — CHANGES_REQUESTED** [review]

File: saagieapi/jobs/jobs.py (hunk 1 above)
Comment (guillaumeprevost76) [reply saagieapi/jobs/jobs.py:896]: import is a reserved world by Python, I can't name the function import.
Maybe you know some tricks to do that

File: saagieapi/jobs/jobs.py (hunk 2 above)
Comment (guillaumeprevost76) [reply saagieapi/jobs/jobs.py:920]: changes done

File: pyproject.toml [hunk 3]
```diff
@@ -24,6 +24,7 @@ pylint = "^2.13.7"
 isort = "^5.10.1"
 black = "^22.3.0"
 pre-commit = "^2.18.1"
+ipykernel = "^6.15.1"
```
Comment (tquiviger) [inline pyproject.toml:27]: useful?

File: saagieapi/jobs/jobs.py (hunk 1 above)
Comment (tquiviger) [reply saagieapi/jobs/jobs.py:896]: import_from_json?

File: saagieapi/jobs/jobs.py [hunk 4]
```diff
@@ -892,3 +892,83 @@ def export(
...
+            jobScheduleTimezone = job_info["scheduleTimezone"]
+            jobResources = job_info["resources"]
+            jobEmails = ""
+            jobStatusList = ""
+
+            if job_info["alerting"] is not None:
+                jobEmails = job_info["alerting"]["emails"]
+                jobStatusList = job_info["alerting"]["statusList"]
+
+            self.create(
```
Comment (tquiviger) [inline saagieapi/jobs/jobs.py:950]: create_or_upgrade instead?

File: pyproject.toml [hunk 5]
```diff
@@ -24,6 +24,7 @@ pylint = "^2.13.7"
...
```
[the remark this answers is not in the document]
Comment (guillaumeprevost76) [reply pyproject.toml:28]: Not necessary, I removed it from the file

File: pyproject.toml (hunk 3 above)
Comment (guillaumeprevost76) [reply pyproject.toml:27]: Not necessary, I removed it from the file

**Review by Qiwei08 — APPROVED** [review]

File: saagieapi/pipelines/pipelines.py [hunk 6]
```diff
@@ -401,6 +401,101 @@ def edit(
...
+                schedule_timezone,
+            )["editPipeline"]
+
+            responses["addGraphPipelineVersion"] = self.upgrade(pipeline_id, graph_pipeline, release_note)[
+                "addGraphPipelineVersion"
+            ]
+
+            return responses
+
+        else:
```
Comment (Qiwei08) [inline saagieapi/pipelines/pipelines.py:486]: Unnecessary "else" after "return"

File: tests/integration/saagie_api_integration_test.py [hunk 7]
```diff
@@ -1,4 +1,5 @@
 # pylint: disable=attribute-defined-outside-init
+import json
```
Comment (Qiwei08) [inline tests/integration/saagie_api_integration_test.py:2]: Unnecessary import json

Comment (guillaumeprevost76) [reply tests/integration/saagie_api_integration_test.py:2]: you're rigth, I remove the import

File: saagieapi/pipelines/pipelines.py (hunk 6 above)
Comment (guillaumeprevost76) [reply saagieapi/pipelines/pipelines.py:486]: This code comes from a rebase and had been done on the feature create_or_upgrade pipeline.
By the way, I'm not agree with your comment, the "else" to separate the create choice and "edit" choice.

File: saagieapi/jobs/jobs.py (hunk 4 above)
Comment (guillaumeprevost76) [reply saagieapi/jobs/jobs.py:959]: As discussed in Tech time, import_from_json is a function made to create (not upgrade) job by using a JSON format (like the one used by the export function).

**Review by TFellet — APPROVED** [review]

**Pull request merged**